manifests/params.pp

```diff
@@ -179,13 +192,16 @@
 
   # Passenger config
   $server_app_root = "${dir}/rack"
-  $server_ssl_dir  = "${vardir}/ssl"
+  $server_ssl_dir  = $ssldir
 
   $server_package     = undef
   $server_version     = undef
-  $client_package     = $::osfamily ? {
-    'Debian' => ['puppet-common','puppet'],
-    default  => ['puppet'],
+  $client_package     = versioncmp($::puppetversion, '4.0') ? {
+    '-1'     => $::osfamily ? {
+      'debian' => ['puppet-common','puppet'],
+      default   => ['puppet'],
+    },
+    default => 'puppet-agent',
```

Comment:
There is no Puppet 4 package for FreeBSD, yet, but I'm not sure if it would be called "puppet-agent"... Anyway, I know this is not ideal, but to be able to also use non-AIO packages on Linux I'd like to have a variable like ```aio_package```, which could be ```true``` for 4.x by default, but setable, so native 4.x packages are not that problematic.